### PR TITLE
#330: Editing playlist's details dont open edit modal

### LINKED ISSUE #330: Editing playlist's details dont open edit modal
**Describe the bug**
Editing a playlist content from the Playlist page (see attached screenshot) redirects to Home page instead of opening modal for editing playlist's details

**To Reproduce**

1. Create a playlist
2. Click the created playlist card on the sidebar
3. Click on three dots and then select `Editar datos`

**Expected behavior**

Edit playlist's details should be opened

**Screenshots**

Image

**Additional context**

---

__🖐️ Learn [how to CONTRIBUTE](https://antoniomrtz.github.io/SpotifyElectron_Web/docs/CONTRIBUTING/), all help is welcomed :)__

__📚 Check out the [project documentation](https://antoniomrtz.github.io/SpotifyElectron_Web/docs/) for more info__

__⭐ Help this project grow by giving it a star if you like it!__

__😵 Trello links are used for traceability, do not ask to join the board before leaving a comment in the related issue__

### PR DESCRIPTION
**Pull request by Chubbi-Stephen**
- Fix route mismatch in ContextMenuPlaylist.tsx
- Change navigation from /playlists/ to /playlist/ to match App.tsx route definition
- Add test case for edit playlist route navigation
- Add localStorage mock for proper test execution
- Resolves issue where 'Editar datos' redirected to home instead of opening edit modal

Changes:
- Modified: ContextMenuPlaylist.tsx - Fixed route navigation (1 line)
- Modified: ContextMenuPlaylist.test.tsx - Added test case and localStorage mock

Tested: Route navigation, localStorage handling, and close handler execution

## Description

Fix playlist edit modal redirecting to Home page instead of opening the edit modal when users click "Editar datos" from the playlist context menu. The issue was caused by a route mismatch where the context menu was navigating to /playlists/ (plural) but the app router only defines /playlist/ (singular).

## Checklist

- [X] I have read and followed the [Contributing Guidelines](https://github.com/AntonioMrtz/SpotifyElectron/blob/master/docs/CONTRIBUTING.md).
- [X] I have followed the project's [git convention](https://github.com/AntonioMrtz/SpotifyElectron/blob/master/docs/developer/Git-Convention.md).
- [X] I have tested the changes and verified they work as intended. Automated tests (unit, integration, e2e as applicable) and manual tests (UI interactions, edge cases, error states, etc.).
- [X] I have set the pull request base to the appropriate `development-vx.xx` branch corresponding to the issue milestone.

## Commit type

FIX
- `feat`: indicates the addition of a new feature or functionality to the project.
- `fix`: used when fixing a bug or error in the code.
- `docs`: changes to the documentation.
- `style`: formatting, missing semi colons, etc; no production code change.
- `refactor`: refactoring production code, eg. renaming a variable.
- `test`: adding missing tests, refactoring tests; no production code change.
- `chore`: updating grunt tasks etc; no production code change.
- `ci`: updating scripts for continuous integration.
- `build`: update building scripts or Docker Images.
- `perf`: update code for performance improvement.
- `revert`: revert changes.
- `release`: only for release versions.
-->

## Issue

closes #330

## Proposed Changes

Fixed route navigation in ContextMenuPlaylist.tsx - Changed navigation from /playlists/${playlistName}?edit=true to /playlist/${playlistName}?edit=true to match App.tsx route definition (line 232)
Added comprehensive test case - Added test for edit playlist route navigation in existing ContextMenuPlaylist.test.tsx to verify correct route is called
Added localStorage mock - Implemented localStorage mock in test file to properly test localStorage flag setting and retrieval functionality
## Potential Impact

✅ Positive Impact
User Experience

Immediate Fix: Users can now successfully edit playlist details without being redirected to Home page
Improved Workflow: Playlist management functionality works as originally intended
Reduced Frustration: Eliminates confusing redirect behavior that broke user expectations
Code Quality

Route Consistency: All frontend playlist routes now follow the same /playlist/ pattern
Better Maintainability: Consistent routing makes the codebase easier to understand and maintain
Enhanced Testing: Added test coverage for playlist context menu functionality
🔍 Risk Assessment
Low Risk Changes

Minimal Code Change: Only 1 line modified in production code
No API Changes: Backend endpoints remain unchanged (/playlists/ for API)
No Breaking Changes: Existing functionality continues to work normally
Backward Compatible: No impact on existing user data or saved states
Potential Side Effects

None Identified: The change only affects the specific broken navigation path
Route Pattern: Change aligns with existing patterns, no conflicts expected
Browser History: Uses replace: true to avoid navigation history issues
🎯 Project Impact
Immediate Benefits

Bug Resolution: Fixes a critical user-facing issue
Feature Completion: Playlist editing feature now works end-to-end
User Retention: Prevents user frustration from broken functionality
Long-term Benefits

Code Consistency: Establishes clear routing patterns for future development
Test Coverage: Provides foundation for testing similar components
Documentation: Creates reference for handling similar routing issues
⚠️ Considerations
Deployment

Zero Downtime: Change can be deployed without service interruption
No Database Changes: No migrations or data updates required
Rollback Ready: Simple to revert if unexpected issues arise
Monitoring

User Behavior: Monitor playlist edit usage to confirm fix effectiveness
Error Rates: Watch for any new navigation-related errors (unlikely)
🚀 Overall Assessment
Impact Level: Low Risk, High Value

Minimal code change with maximum user benefit
No architectural changes or breaking modifications
Well-tested and thoroughly verified solution

## Screenshots

spotifyElectron Test

## Assigned

@AntonioMrtz

### CODE REVIEW

**Comment by Chubbi-Stephen** [comment]
@AntonioMrtz Please can I get a review on this PR?

File: Electron/src/__tests__/components/ContextMenu/ContextMenuPlaylist.test.tsx [hunk 1]
```diff
@@ -195,3 +219,40 @@ test('ContextMenuPlaylist Add Playlist to Playlist', async () => {
...
+  });
+
+  // Find and click the edit button
+  const editButton = component.getByText(t('contextMenuPlaylist.edit'));
+
+  await act(async () => {
+    fireEvent.click(editButton);
+  });
+
+  // Verify navigation was called with correct route (singular /playlist/, not /playlists/)
```
Comment (AntonioMrtz) [inline Electron/src/__tests__/components/ContextMenu/ContextMenuPlaylist.test.tsx:250]: We can remove this comment

**Comment by AntonioMrtz** [comment]
The code looks good. Can you add yourself to contributors and readme file? :)

**Review by AntonioMrtz — CHANGES_REQUESTED** [review]
Previous comments

**Comment by AntonioMrtz** [comment]
HI @Chubbi-Stephen can you add yourself as contributor in `README` and `CONTRIBUTORS`?

**Review by AntonioMrtz — CHANGES_REQUESTED** [review]

**Comment by AntonioMrtz** [comment]
Inactivity
